docs2/site/docs/analyzers/GQL004_DoNotUseObsoleteFieldMethods.md

```diff
@@ -0,0 +1,131 @@
+
+## Example of a violation
+
+```c#
+Field<NonNullGraphType<StringGraphType>>(
+    "name",
+    "Field description",
+    resolve: context => context.Source!.Name);
+
+FieldAsync<CharacterInterface>(
+    "hero",
+    resolve: async context => await data.GetDroidByIdAsync("3"));
+
+FieldAsync<HumanType>(
+    "human",
+    arguments: new QueryArguments(new QueryArgument<NonNullGraphType<StringGraphType>>
+    {
+        Name = "id",
+        Description = "id of the human"
+    }),
+    resolve: async context => await data.GetHumanByIdAsync(context.GetArgument<string>("id"))
+);
+```
+
+## Example of how to fix
+
+```c#
+Field<NonNullGraphType<StringGraphType>>("name")
+    .Description("Field description")
+    .Resolve(context => context.Source!.Name);
+
+Field<CharacterInterface>("hero")
+    .ResolveAsync(async context => await data.GetDroidByIdAsync("3"));
+
+Field<HumanType>("human")
+    .Argument<NonNullGraphType<StringGraphType>>("id", "id of the human")
+    .ResolveAsync(async context => await data.GetHumanByIdAsync(context.GetArgument<string>("id")));
+
+```
+
+## Suppress a warning
+
+If you just want to suppress a single violation, add preprocessor directives to your source file to disable and then re-enable the rule.
+
+```csharp
+#pragma warning disable GQL004
+// The code that's violating the rule is on this line.
+#pragma warning restore GQL004
+```
+
+To disable the rule for a file, folder, or project, set its severity to `none` in the [configuration file](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files).
+
+```ini
+[*.cs]
+dotnet_diagnostic.GQL004.severity = none
+```
+
+For more information, see [How to suppress code analysis warnings](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/suppress-warnings).
+
+## Configure code fix
+
+The given diagnostic rule offers an automatic code fix. By default, it attempts to retain the original user code formatting as much as possible. For instance, consider the subsequent code snippet:
+
+```c#
+Field<StringGraphType>("name", "description", null,
+    context => "text");
+```
+
+This will be transformed into:
+
+```c#
+Field<StringGraphType>("name").Description("description").Arguments(null)
+    .Resolve(context => "text");
+```
+
+### Configure formatting
+
+The `reformat` configuration option will guide the code fix to apply code reformatting:
+
+```ini
+[*.cs]
+dotnet_diagnostic.GQL004.reformat = true
+```
+
+The earlier code snippet will undergo the following reformatting:
+
+```c#
+Field<StringGraphType>("name")
+    .Description("description")
+    .Arguments(null)
+    .Resolve(context => "text");
+```
+
+### Configure `null` values handling
+
+The `skip_nulls` option can be used to remove unnecessary `null` values assignments
+
+```ini
+[*.cs]
+dotnet_diagnostic.GQL004.skip_nulls = true
+```
```

Comment:
What is default?  Can we configure so `skip_nulls` is `true` by default?  Generally I would expect that null was provided only to skip an argument to the Field method.

Reply:
The `null` could also be provided to the `description` argument. For example
```c#
public class MyGraphType : ObjectGraphType
{
    public MyGraphType()
    {
        Field<StringGraphType>(
            "name",
            null,
            null,
            context => "text");
    }
}
```
But I agree, that removing redundant nulls by default sounds like a better choice. Do you prefer to keep the `skip_nulls` option name or rename it to `preserve_nulls` in that case?

Reply:
> Do you prefer to keep the `skip_nulls` option name or rename it to `preserve_nulls` in that case?

Either way; your call.